includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -4495,7 +4495,33 @@ private function upe_needs_redirection( $payment_methods ) {
 	 * @return void
 	 */
 	private function handle_afterpay_shipping_requirement( WC_Order $order, Create_And_Confirm_Intention $request ): void {
-		$check_if_usable = function ( array $address ): bool {
+		$wc_locale_data = WC()->countries->get_country_locale();
+
+		$check_if_usable = function ( array $address ) use ( $wc_locale_data ): bool {
+			if ( $address['country'] && isset( $wc_locale_data[ $address['country'] ] ) ) {
+				$country_locale_data = $wc_locale_data[ $address['country'] ];
+				$fields_to_check     = [
+					'state'     => 'state',
+					'city'      => 'city',
+					'postcode'  => 'postal_code',
+					'address_1' => 'line1',
+				];
+
+				foreach ( $fields_to_check as $locale_field => $address_field ) {
+					$is_field_required = (
+						! isset( $country_locale_data[ $locale_field ] ) ||
+						! isset( $country_locale_data[ $locale_field ]['required'] ) ||
+						false !== $country_locale_data[ $locale_field ]['required']
+					);
```

Comment:
```suggestion
					$is_field_required = (
						! isset( $country_locale_data[ $locale_field ] ) ||
						! isset( $country_locale_data[ $locale_field ]['required'] ) ||
						$country_locale_data[ $locale_field ]['required']
					);
```
I suppose we could rewrite this last conditional as the above, since I think we can usually assume that `$country_locale_data[ $locale_field ]['required']` will be a boolean value. Though, I agree that as things as your statement is probably more explicit and safer in case someone else injects some junk values into `woocommerce_get_country_locale`. 🤷